### PR TITLE
Create CipherEX.md

### PR DESCRIPTION
**Pull request by pawnz0**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum (DAI) address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we've successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Could you share some of your previous (open source) work with us? And isn’t nucypher already working on the [rust implementation of Umbral](https://github.com/nucypher/rust-umbral)? To be honest, I find it a little bit strange that you don’t even mention this in your application, since your first milestone sounds exactly like what nucypher has already done. But I might be missing something here.

File: applications/CipherEX.md
Comment (burdges) [inline applications/CipherEX.md:25]: Ristretto is Ed25519 under the hood, no pairings.  We've no pairings on substrate currently, but maybe by the end of the summer.

File: applications/CipherEX.md
Comment (burdges) [inline applications/CipherEX.md:11]: There is nothing here about why threshold proxy reencryption works better than [FairSwap](https://eprint.iacr.org/2018/740.pdf) or [CMFE](https://eprint.iacr.org/2015/064.pdf) or the numerous other fare exchange protocols.  And proxy reencryption obviously comes with major downsides.

**Comment by burdges** [comment]
I've never quite understood what NuCyoher does, despite chats with @tuxxy, but mostly they never discuss one-off data delivery, so presumably simple protocols work better there.  It's plausible we'd become an attractive platform for them once we land pairing friendly crypto, but..

We've fundamentally avoided threshold security models in polkadot because being large enough for security but small enough that thresholds work correctly becomes messy, not merely technically but via community management, etc.  It's plausible our NPoS system makes substrate somewhat more friendly for the community management side, except you'd want annoying subcommittee crap similar to [RandHerd](https://eprint.iacr.org/2016/1067.pdf)/[drand](https://github.com/drand/drand).

Anyone doing threshold schemes on parachains must solve these issues themselves, so maybe easiest to bridge to NuCipher's chain first, which may fall under some existing bridge work plan.

**Comment by tuxxy** [comment]
Hey, thanks for the tag @burdges!

Happy to chat about any collaboration. We've been actively exploring integrations for both Polkadot and Tendermint. Feel free to reach out to tux@nucypher.com and we can maybe take that convo there!

Happy Easter!

**Comment by alxs** [comment]
@tuxxy if you're considering different options for a first grant based on Jeff's comments, we can put this on hold or you can close the application for now.

Otherwise a reminder to also address David's comments:

> Could you share some of your previous (open source) work with us? And isn’t nucypher already working on the rust implementation of Umbral? To be honest, I find it a little bit strange that you don’t even mention this in your application, since your first milestone sounds exactly like what nucypher has already done. But I might be missing something here.

**Comment by pawnz0** [comment]
Hello, we want to clarify that we are not Nucypher team. I am currently a Cryptography PHD student UCAS. And my teammate work as a developer in a blockchain company. Here is his Github [link](https://github.com/TerenceClark). And we are not aware of this Umbral-rust implementation. Thanks for your reminding, we will remove the milestone 1 from our proposal. We would fork Nucypher's Repo and made some necessary modification such as 25519 curve support based on this [Repo](https://github.com/nucypher/rust-umbral).

Also we would elaborate how we select the Proxy in our project. First, we will use NPOS mechanism to elect N collators(collator is also Proxy) in each Epoch. When Alice need to authorize Bob the access to the secret, she can divide the re-cap key into N pieces of fragment using VSS. Alice could sent one key fragment to each collator. In our setting, we can guarantee that whole procedure would work properly even N-t collator are malfunction.

**Comment by tuxxy** [comment]
@pawnz0 I can't find contact information for you, but we'd love to chat about NuCypher interop and maybe help out with grants as well.

Feel free to email tux@nucypher.com, and we can go from there. :)

**Comment by pawnz0** [comment]
@tuxxy  Thanks. I will reach out to you by email.

**Comment by alxs** [comment]
Thanks for clarifying, makes sense now :)

I lack the necessary background but as is I don't see where this would be useful? And do you have any prior work that you could share, since both you and @TerenceClark's profiles are pretty empty?

**Comment by pawnz0** [comment]
@alxs 
Thanks for your reply. As a PHD(pursuing) student, I dont have much experience in open source project. I can share some information about my previous work.  I am currently a member of State Key Laboratory of Information Security, Institute of Information Engineering (IIE), Chinese Academy of Sciences. And I have participate some crypto competitions and workshop. Here are some in the list:  "FinanceCrypto"2020 China National Cryptography Competition《A supervised scheme of data privacy and identity anonymity》, 2020 China National Crypto-Math Challenge《Subset Sum Problem》, 2020 China National Post-Graduate Mathematical Contest in Modeling《UAV cluster cooperative confrontation》, The 6th Workshop on Cyberspace-FHE Application＆its Standardization and The 2nd Crypto Innovation School.  And my teammate Terence have participated in a blockchain program called [Dipperin](https://github.com/dipperin/dipperin-core).

**Comment by alxs** [comment]
Thanks @pawnz0. And could you explain to me in layman's terms why this would be useful?

**Comment by pawnz0** [comment]
> Thanks @pawnz0. And could you explain to me in layman's terms why this would be useful?

Hello, could you please close this PR. As we have discussed with Nucipher and we decide to bridge Ursulas network from Ethereum to Polkadot. We will start another proposal later once we finish the design. Thanks.

**Comment by alxs** [comment]
Sounds good.

**Pull request closed without merging**
